app/src/main/java/com/example/pokedexdesafio/data/model/PokemonDetailResponse.kt

```diff
@@ -16,21 +17,108 @@ data class PokemonDetail(
 
 //Type
 @Parcelize
-data class TypeContainer(val slot: Int, val type: Type) : Parcelable
+data class TypeContainer(val slot: Int?, val type: Type?) : Parcelable
 
 @Parcelize
-data class Type(val name: String, val url: String) : Parcelable
+data class Type(val name: String?, val url: String?) : Parcelable
 
 //Attacks
 @Parcelize
-data class MoveContainer(val move: Move) : Parcelable
+data class MoveContainer(val move: Move?) : Parcelable
 
 @Parcelize
-data class Move(val name: String, val url: String) : Parcelable
+data class Move(val name: String?, val url: String?) : Parcelable
 
 //Abilities
 @Parcelize
-data class AbilityContainer(val ability: Ability, val isHidden: Boolean, val slot: Int) : Parcelable
+data class AbilityContainer(val ability: Ability?, val isHidden: Boolean?, val slot: Int?) : Parcelable
 
 @Parcelize
-data class Ability(val name: String, val url: String) : Parcelable
+data class Ability(val name: String?, val url: String?) : Parcelable
+
+// Ext functions for mapping into business models
+fun Ability.toAbility(): AbilityDetail? =
+    if (name.isNullOrBlank() || url.isNullOrBlank()) {
+        null
+    } else {
+        AbilityDetail(name, url)
+    }
+
+fun AbilityContainer.toAbilityList(): AbilityList? =
+    if(ability != null || isHidden != null || slot != null){
+        AbilityList(ability?.toAbility(), isHidden, slot)
+    }else{
+        null
+    }
+fun PokemonDetailResponse.toAbilitiesList(): List<AbilityList> = abilities.mapNotNull { it?.toAbilityList() }
```

Comment:
AbilityList should be renamed with a meaningful name

Reply:
solved